projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -126,33 +140,264 @@ export function defineProperty(
+
+    const constructorParameterTypeReferenceNode = constructorParameter
+      .getChildren()
+      .find(node => node.kind === ts.SyntaxKind.TypeReference);
+    if (!constructorParameterTypeReferenceNode) {
+      return false;
+    }
+    const constructorParameterType = findLevel1NodesByTextAndKind(
+      constructorParameterTypeReferenceNode.getChildren(),
+      parameterClassType.className,
+      ts.SyntaxKind.Identifier
+    );
+
+    // return false if there's no param with the expected type on the current position
+    if (constructorParameterType.length === 0) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function checkSuper(
+  constructorNode: ts.Node,
+  parameterClassTypes: ClassType[]
+): boolean {
+  const callExpressions = findNodes(
+    constructorNode,
+    ts.SyntaxKind.CallExpression
+  );
+  if (callExpressions.length === 0) {
+    return false;
+  }
+  // super has to be the first expression in constructor
+  const firstCallExpression = callExpressions[0];
+  const superKeyword = findNodes(
+    firstCallExpression,
+    ts.SyntaxKind.SuperKeyword
+  );
+  if (superKeyword && superKeyword.length === 0) {
+    return false;
+  }
+
+  const params = findNodes(firstCallExpression, ts.SyntaxKind.Identifier);
+  if (params.length !== parameterClassTypes.length) {
+    return false;
+  }
+
+  return true;
+}
+
+export function addConstructorParam(
+  source: ts.SourceFile,
+  sourcePath: string,
+  constructorNode: ts.Node | undefined,
+  paramToAdd: ClassType
+): Change[] {
+  if (!constructorNode) {
+    throw new SchematicsException(`No constructor found in ${sourcePath}.`);
+  }
+
+  const changes: Change[] = [];
+
+  changes.push(
+    injectService(
+      constructorNode,
+      sourcePath,
+      paramToAdd.className,
+      'no-modifier'
+    )
+  );
+
+  if (!isImported(source, paramToAdd.className, paramToAdd.importPath)) {
+    changes.push(
+      insertImport(
+        source,
+        sourcePath,
+        paramToAdd.className,
+        paramToAdd.importPath
+      )
+    );
+  }
+
+  changes.push(
+    updateConstructorSuperNode(
+      sourcePath,
+      constructorNode,
+      paramToAdd.className
+    )
+  );
+
+  return changes;
+}
+
+function updateConstructorSuperNode(
+  sourcePath: string,
+  constructorNode: ts.Node,
+  propertyName: string
+): InsertChange {
+  const callExpressions = findNodes(
+    constructorNode,
+    ts.SyntaxKind.CallExpression
+  );
+  propertyName = strings.camelize(propertyName);
+
+  if (callExpressions.length === 0) {
+    throw new SchematicsException('No super() call found.');
+  }
+  // super has to be the first expression in constructor
+  const firstCallExpression = callExpressions[0];
+  const superKeyword = findNodes(
+    firstCallExpression,
+    ts.SyntaxKind.SuperKeyword
+  );
+  if (superKeyword && superKeyword.length === 0) {
+    throw new SchematicsException('No super() call found.');
+  }
+
+  let toAdd = '';
```

Comment:
Can we change it to a more descriptive name?

Reply:
Changed to `toInsert`, let me know if you think it should be more descriptive